### DISCUSSION TITLE
OIDC: oidc not working on grafana accessible via multiple domain names

### DISCUSSION THREAD

**Discussion by robert-heinzmann-logmein** [grafana/grafana#39098] [Platform]
**What happened**:

When grafana is accessible via multiple domain names (e.g. publically and internally without split horizon DNS) and having configured OIDC as an authentication source, OIDC authentication does not work if you set GF_SERVER_ENFORCE_DOMAIN=false (default). Enabeling GF_SERVER_ENFORCE_DOMAIN=true, breaks the multi path access (e.g. public and internal via different hostnames) and thus makes this deployment option impossible.

The error is not related to the OIDC setting, as the oidc app has been configured with multiple allowed redirect urls (public and privat)

**What you expected to happen**:

Make OIDC work with Grafana indipendent of the hostname used to access grafana. (Basically make it work for all domain names configured in DNS with GF_SERVER_ENFORCE_DOMAIN=false)

**How to reproduce it (as minimally and precisely as possible)**:

- Configure 2 DNS entries for grafana (e.g. grafana-public.example.com and grafana-private.example.com)
- Configure your OIDC provider to allow both redirect URLS (e.g. Azure AD to allow grafana-public.example.com/login/generic_oauth and grafana-private.example.com/login/generic_oauth)
- Configure GF_SERVER_ENFORCE_DOMAIN=false (default)
- Configure GF_SERVER_DOMAIN=grafana-public.example.com
- Access grafana via grafana-public.example.com --> Works
- Access grafana via grafana-private.example.com --> Does not Work (see attached Screenshot)

![image](https://user-images.githubusercontent.com/83008166/123270183-f8bfcb00-d4ff-11eb-9736-0d44d651e454.png)

**Anything else we need to know?**:

**Environment**:
- Grafana version: Grafana v7.1.0 (8101355285)
- Data source type & version: Various
- OS Grafana is installed on: Docker Image
- User OS & Browser: Chrome / Firefox
- Grafana plugins: Lots
- Others: OIDC configured via AAD as generic_oauth

**Comment by robert-heinzmann-logmein** [comment]
Hi there, any progress on this issue so far ?

**Comment by marefr** [comment]
Only one domain is currently supported and we have no current plan to support more. Would suggests looking into using a proxy to be able to use two domains, not sure possible but this [post](https://community.grafana.com/t/multiple-root-urls-for-grafana/6562/4) indicate it should be possible.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
